src/library_pthread.js

```diff
@@ -154,7 +154,18 @@ var LibraryPThread = {
       }
       PThread.runningWorkers = [];
     },
+#if USE_ASAN || USE_LSAN
+    freeThreadData__deps: ['$withBuiltinMalloc'],
+#endif
```

Comment:
Is this an essential part of this change?

Reply:
Did you mean this `__deps` line? Or the `withBuiltinMalloc` in general? The comment above `withBuiltinMalloc` should explain why that is necessary, but perhaps it can be elaborated on further.

(Perhaps this could be moved to native code and mark that function with the `no_sanitize("address")` attribute(?))

Reply:
The __deps line is necessary, but none of the conditionals should be needed.

Reply:
Ah, I had to move the `$withBuiltinMalloc` to `$PThread__deps` instead, otherwise it caused this exception when building without sanitizers:
```
exception thrown: ReferenceError: withBuiltinMalloc is not defin
```